ibm/resource_ibm_pi_placement_group.go

```diff
@@ -0,0 +1,153 @@
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	models "github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+)
+
+const (
+	PIInstancePlacementGroupID = "pi_instance_placement_group_id"
+	PIPlacementGroupMembers    = "pi_placement_group_members"
+)
+
+func resourceIBMPIPlacementGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPIPlacementGroupCreate,
+		ReadContext:   resourceIBMPIPlacementGroupRead,
+		UpdateContext: resourceIBMPIPlacementGroupUpdate,
+		DeleteContext: resourceIBMPIPlacementGroupDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			helpers.PIPlacementGroupName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the placement group",
+			},
+
+			helpers.PIPlacementGroupPolicy: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Policy of the placement group",
+			},
+
+			helpers.PICloudInstanceId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PI cloud instance ID",
+			},
+
+			PIPlacementGroupMembers: {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Server IDs that are the placement group members",
+			},
+
+			helpers.PIPlacementGroupID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "PI placement group ID",
+			},
+		},
+	}
+}
+
+func resourceIBMPIPlacementGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
+	name := d.Get(helpers.PIPlacementGroupName).(string)
+	policy := d.Get(helpers.PIPlacementGroupPolicy).(string)
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, cloudInstanceID)
+	body := &models.PlacementGroupCreate{
+		Name:   &name,
+		Policy: &policy,
+	}
+
+	response, err := client.Create(body)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.FromErr(err)
+	}
+
+	log.Printf("Printing the placement group %+v", &response)
+
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *response.ID))
+	return resourceIBMPIPlacementGroupRead(ctx, d, meta)
+}
+
+func resourceIBMPIPlacementGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to get the placement group %v", err)
```

Comment:
This is not GET failure. You can directly return the error.

Reply:
Done.